File: android/chat-client/src/main/kotlin/com/anapaya/chat/client/Transport.kt
// Copyright 2026 Anapaya Systems

package com.anapaya.chat.client

/** One request, as the chat API asks for it. */
public data class ChatRequest(
    val method: String,
    /** Relative to the server's `/api/v1`, with a leading slash. */
    val path: String,
    val json: String? = null,
    /** The session token, on the calls that need one. */
    val bearer: String? = null,
)

/** What came back. The body is text: every reply this API sends is JSON. */
public data class ChatReply(
    val status: Int,
    val body: String,
)

/** Puts a request on the wire and brings the reply back. An interface, so a test can fake it. */
public interface Transport {
    public suspend fun send(request: ChatRequest): ChatReply

    /** Releases the connections. The client cannot be used afterwards. */
    public fun close()
}

File: ios/ChatClient/Sources/ChatClient/Transport.swift
// Copyright 2026 Anapaya Systems

import Foundation

/// One request, as the chat API asks for it.
public struct ChatRequest: Sendable {
    public let method: String
    /// Joined onto the server's base URL and its API prefix.
    public let path: String
    public let json: String?
    public let bearer: String?

    public init(method: String, path: String, json: String? = nil, bearer: String? = nil) {
        self.method = method
        self.path = path
        self.json = json
        self.bearer = bearer
    }
}

/// What came back.
public struct ChatReply: Sendable {
    public let status: Int
    public let body: String

    public init(status: Int, body: String) {
        self.status = status
        self.body = body
    }
}

/// Puts a request on the wire and brings the reply back. A protocol, so a test can fake it.
public protocol Transport: Sendable {
    func send(_ request: ChatRequest) async throws -> ChatReply
    func close() async
}
